lib/aws/rails/middleware/ebs_sqs_active_job_middleware.rb

```diff
@@ -85,7 +85,15 @@ def sent_from_docker_host?(request)
       end
 
       def app_runs_in_docker_container?
-        @app_runs_in_docker_container ||= `[ -f /proc/1/cgroup ] && cat /proc/1/cgroup` =~ /docker/
+        @app_runs_in_docker_container ||= in_docker_containr_with_cgroup1? || in_docker_containr_with_cgroup2?
+      end
+
+      def in_docker_containr_with_cgroup1?
+        File.exist?('/proc/1/cgroup') && File.read('/proc/1/cgroup') =~ %r{/docker/}
+      end
+
+      def in_docker_containr_with_cgroup2?
+        File.exist?('/proc/1/mountinfo') && File.read('/proc/1/mountinfo') =~ %r{/docker/containers/}
```

Comment:
This [article](https://www.baeldung.com/linux/is-process-running-inside-container) suggests `/proc/self/mountinfo`

Reply:
Thanks for reviewing.
Fixed in the following commit.
https://github.com/aws/aws-sdk-rails/pull/116/commits/845580bcfc4a092d28ba5a2f245112c272d151d9